### PR TITLE
clang-tidy: Apply modernize-use-equals-default everywhere

### PR DESCRIPTION
**Pull request by irrenhaus3**
This replaces all empty or trivial implementations of constructors, destructors and assignment operators with `= default;`, explicitly stating that these functions should be automatically generated by the compiler.  
I also ran `modernize-use-equals-delete` on everything, but this only hit 3rdparty submodules. Our code got 0 flags.

### CODE REVIEW

File: include/MidiEvent.h [hunk 1]
```diff
@@ -66,16 +66,7 @@ class MidiEvent
...
-		m_type( other.m_type ),
-		m_metaEvent( other.m_metaEvent ),
-		m_channel( other.m_channel ),
-		m_data( other.m_data ),
-		m_sysExData( other.m_sysExData ),
-		m_sourcePort(other.m_sourcePort),
-		m_source(other.m_source)
-	{
-	}
+	MidiEvent( const MidiEvent& other ) = default;
```
Comment (irrenhaus3) [inline include/MidiEvent.h:69]: Particularly impressed by this one here.

File: include/DetuningHelper.h [hunk 2]
```diff
@@ -43,8 +43,7 @@ class DetuningHelper : public InlineAutomation
 	}
 
 	~DetuningHelper() override
-	{
-	}
+	= default;
```
Comment (JohannesLorenz) [inline include/DetuningHelper.h:46]: Many lines contain only `= default` now. Grep those and move them behind the actual function?

Comment (irrenhaus3) [reply include/DetuningHelper.h:46]: Done.

File: include/DummyEffect.h [hunk 3]
```diff
@@ -61,8 +61,7 @@ class DummyEffectControls : public EffectControls
 	}
 
 	~DummyEffectControls() override
-	{
-	}
+	= default;
```
Comment (JohannesLorenz) [inline include/DummyEffect.h:64]: Nope?

Comment (irrenhaus3) [reply include/DummyEffect.h]: No idea what you're talking about. :smirk:

**Review by JohannesLorenz — APPROVED** [review]

**Comment by JohannesLorenz** [comment]
Basic tests also successfull.

**Review by DomClark — APPROVED** [review]

**Pull request merged**